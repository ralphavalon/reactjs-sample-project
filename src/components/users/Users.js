import React from 'react';
import './Users.css';

function Users(props) {
  return (
    <div className="App">
      <header className="App-header">
        <ul>
            {props.users.map((user) => (
                <li key={user.id}>{user.name} - {user.age} - {user.city}</li>
            ))}
        </ul>
      </header>
    </div>
  );
}

export default Users;
