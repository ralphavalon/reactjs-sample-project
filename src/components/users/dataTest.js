export const users = [{
    id: 1,
    name: 'Ralph Avalon',
    city: 'Rio de Janeiro',
    age: 13
}];
