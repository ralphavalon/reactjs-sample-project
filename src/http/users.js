import axios from 'axios';

export const loadUsers = () => {
    // axios.get('http://localhost:8081/rest')
    return axios.get('http://run.mocky.io/v3/c1ba1a88-b5f1-4dd5-95e2-9ec01d25d577');
}

export default loadUsers;
