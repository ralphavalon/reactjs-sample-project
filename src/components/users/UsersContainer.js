import React from 'react';
import http from '../../http';
import Users from './Users';
import './Users.css';

class UsersContainer extends React.Component {
    constructor(props) {
        super(props);
        this.state = { users: [] };
    }

    componentDidMount() {
        http.loadUsers().then(response => {
            this.setState({ users: response.data });
        });
    }

    render() {
        return React.createElement(Users, {
            users: this.state.users,
            ...this.props
        });
    }
}

export default UsersContainer;
