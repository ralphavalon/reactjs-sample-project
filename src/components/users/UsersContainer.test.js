import React from 'react';
import { render } from '@testing-library/react';
import users from '../../http';
import * as dataTest from './dataTest';
import UsersContainer from './UsersContainer';

jest.mock('../../http');

test('renders learn react link', async () => {
  users.loadUsers.mockResolvedValue({ data: dataTest.users });

  const { container } = render(<UsersContainer />);
  await Promise.resolve();

  expect(container).toMatchSnapshot();
});
