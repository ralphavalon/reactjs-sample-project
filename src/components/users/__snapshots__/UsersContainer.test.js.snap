// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <ul>
        <li>
          Ralph Avalon
           - 
          13
           - 
          Rio de Janeiro
        </li>
      </ul>
    </header>
  </div>
</div>
`;
